finality-aleph/src/party/manager/authority.rs

```diff
@@ -65,31 +67,39 @@ impl Subtasks {
         // both member and aggregator are implicitly using forwarder,
         // so we should force them to exit first to avoid any panics, i.e. `send on closed channel`
         debug!(target: "aleph-party", "Started to stop all tasks");
-        self.member.stop().await;
+        if self.member.stop().await.is_err() {
+            warn!(target: "aleph-party", "Member stopped with en error");
+        }
         trace!(target: "aleph-party", "Member stopped");
-        self.aggregator.stop().await;
+        if self.aggregator.stop().await.is_err() {
+            warn!(target: "aleph-party", "Aggregator stopped with en error");
+        }
         trace!(target: "aleph-party", "Aggregator stopped");
-        self.refresher.stop().await;
+        if self.refresher.stop().await.is_err() {
+            warn!(target: "aleph-party", "Refresher stopped with en error");
+        }
         trace!(target: "aleph-party", "Refresher stopped");
-        self.data_store.stop().await;
+        if self.data_store.stop().await.is_err() {
+            warn!(target: "aleph-party", "DataStore stopped with en error");
+        }
         trace!(target: "aleph-party", "DataStore stopped");
     }
 
     /// Blocks until the task is done and returns true if it quit unexpectedly.
     pub async fn failed(mut self) -> bool {
```

Comment:
Suggestion: You could change this to also return a `Result<(), ()>` - a `Result` is self-documenting to an extent, while a `bool` requires some extra knowledge of which value means what.

Reply:
changed to `fn wait_completion(mut self) -> Result<(), ()>`